File: app/src/main/java/com/adrzdv/mtocp/domain/repository/TrainRepository.java
package com.adrzdv.mtocp.domain.repository;

import com.adrzdv.mtocp.data.db.entity.TrainEntity;
import com.adrzdv.mtocp.domain.model.revisionobject.collectors.TrainDomain;

import java.util.List;

public interface TrainRepository {

    List<TrainEntity> getAll();

    TrainDomain getTrain(String str);

    void saveAll(List<TrainEntity> entities);
}

File: app/src/main/java/com/adrzdv/mtocp/domain/repository/TrainRepository.kt
package com.adrzdv.mtocp.domain.repository

import com.adrzdv.mtocp.data.db.entity.TrainEntity
import com.adrzdv.mtocp.data.db.pojo.TrainWithDepotAndBranch
import com.adrzdv.mtocp.domain.model.revisionobject.collectors.TrainDomain

interface TrainRepository : BaseAppRepository<TrainEntity, Int> {
    suspend fun getTrainByNumber(nmb: String): TrainDomain
    suspend fun getTrainWithAllData(): List<TrainWithDepotAndBranch>
}